twoliter/embedded/Dockerfile

```diff
@@ -158,18 +176,35 @@ FROM sdk AS repobuild
 ARG PACKAGES
 ARG ARCH
 ARG NOCACHE
-WORKDIR /root
 
+WORKDIR /home/builder
+USER builder
+
+COPY --chown=builder --from=rpm-macros-and-bconds /home/builder/generated.* .
+
+# Build the metadata RPM for the variant.
+RUN --mount=target=/host \
+   cat "/usr/lib/rpm/platform/${ARCH}-bottlerocket/macros" generated.rpmmacros > .rpmmacros \
+   && cat generated.bconds /host/build/tools/metadata.spec >> rpmbuild/SPECS/metadata.spec \
+   && rpmbuild -ba --clean \
+      --undefine _auto_set_build_flags \
+      --define "_target_cpu ${ARCH}" \
+      rpmbuild/SPECS/metadata.spec \
+   && rpm -qp --provides rpmbuild/RPMS/${ARCH}/bottlerocket-metadata-*.${ARCH}.rpm \
+   && echo ${NOCACHE}
+
+WORKDIR /root
 USER root
 RUN --mount=target=/host \
     mkdir -p /local/rpms ./rpmbuild/RPMS \
     && ln -s /host/build/rpms/*.rpm ./rpmbuild/RPMS \
+    && ln -s /home/builder/rpmbuild/RPMS/*/*.rpm ./rpmbuild/RPMS \
     && createrepo_c \
         -o ./rpmbuild/RPMS \
         -x '*-debuginfo-*.rpm' \
         -x '*-debugsource-*.rpm' \
         --no-database \
-        /host/build/rpms \
```

Comment:
Previously it didn't matter whether we used `/host/build/rpms` or `./rpmbuild/RPMS` to generate the repo, but now that we build an RPM in the course of building the variant, we have to use the latter because it's writable.